base/bitarray.jl

```diff
@@ -933,6 +947,7 @@ function _deleteat!(B::BitVector, i::Integer)
 end
 
 function deleteat!(B::BitVector, i::Integer)
+    i = Int(i)
```

Comment:
Could do the "split out the `Int` conversion part"-dance here as well.

Reply:
My feeling is the bit of bounds-checking isn't worth worrying about.